dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/config/TrackerValidationConfig.java

```diff
@@ -50,60 +72,77 @@
-            PreCheckMandatoryFieldsValidationHook.class,
-            PreCheckMetaValidationHook.class,
-            PreCheckUpdatableFieldsValidationHook.class,
-            PreCheckDataRelationsValidationHook.class,
-            PreCheckSecurityOwnershipValidationHook.class,
+            getHookByClass( PreCheckUidValidationHook.class ),
+            getHookByClass( PreCheckExistenceValidationHook.class ),
+            getHookByClass( PreCheckMandatoryFieldsValidationHook.class ),
+            getHookByClass( PreCheckMetaValidationHook.class ),
+            getHookByClass( PreCheckUpdatableFieldsValidationHook.class ),
+            getHookByClass( PreCheckDataRelationsValidationHook.class ),
+            getHookByClass( PreCheckSecurityOwnershipValidationHook.class ),
 
-            TrackedEntityAttributeValidationHook.class,
+            getHookByClass( TrackedEntityAttributeValidationHook.class ),
 
-            EnrollmentNoteValidationHook.class,
-            EnrollmentInExistingValidationHook.class,
-            EnrollmentGeoValidationHook.class,
-            EnrollmentDateValidationHook.class,
-            EnrollmentAttributeValidationHook.class,
+            getHookByClass( EnrollmentNoteValidationHook.class ),
+            getHookByClass( EnrollmentInExistingValidationHook.class ),
+            getHookByClass( EnrollmentGeoValidationHook.class ),
+            getHookByClass( EnrollmentDateValidationHook.class ),
+            getHookByClass( EnrollmentAttributeValidationHook.class ),
 
-            EventCategoryOptValidationHook.class,
-            EventDateValidationHook.class,
-            EventGeoValidationHook.class,
-            EventNoteValidationHook.class,
-            EventDataValuesValidationHook.class,
+            getHookByClass( EventCategoryOptValidationHook.class ),
+            getHookByClass( EventDateValidationHook.class ),
+            getHookByClass( EventGeoValidationHook.class ),
+            getHookByClass( EventNoteValidationHook.class ),
+            getHookByClass( EventDataValuesValidationHook.class ),
 
-            RelationshipsValidationHook.class,
+            getHookByClass( RelationshipsValidationHook.class ),
 
-            AssignedUserValidationHook.class,
+            getHookByClass( AssignedUserValidationHook.class ),
 
-            RepeatedEventsValidationHook.class // This validation must be run
+            getHookByClass( RepeatedEventsValidationHook.class ) // This
+                                                                 // validation
+                                                                 // must be run
         // after
         // all the Event validations
         // because it needs to consider all and only the valid events
         );
     }
 
-    @Bean( "validationOrderMap" )
-    public Map<Class<? extends TrackerValidationHook>, Integer> getValidationOrderMap()
+    private TrackerValidationHook getHookByClass( Class<? extends TrackerValidationHook> hookClass )
```

Comment:
Could we change this method to accept a collection and return a list so we can remove all the getHookByClass calling method repetition?

Reply:
will try 😄

Reply:
Or as I suggest use the original lost and map it using `getHookByClass` as the mapping function.

Reply:
hopefully resolved 😋